src/impy/__init__.py

```diff
@@ -1,18 +1,19 @@
-from .version import __version__  # noqa
-from os.path import join, abspath, dirname
+from .version import __version__
+from pathlib import Path
 from . import config as _config
-
-impy_config = _config.__dict__
-
 from particletools.tables import PYTHIAParticleData
 
-base_path = abspath(dirname(__file__))
+impy_config = _config.__dict__
+base_path = Path(__file__).parent.absolute()
 
 pdata = PYTHIAParticleData(
-    cache_file=open(join(base_path, impy_config["pdata_cachefile"]), "wb")
+    cache_file=open(base_path / impy_config["pdata_cachefile"], "wb")
 )
 
+# these must be after pdata for now
+# FIXME code should not depend on import order (HD)
+from . import models  # noqa
+from . import kinematics  # noqa
+from . import constants  # noqa
 
```

Comment:
I prefer relative imports because the syntax makes it immediately clear that we are importing from our own package and not from third-party. It also avoids the redundant `from impy ...`.

Pep8 recommends absolute imports, but also admits relative imports.  
https://peps.python.org/pep-0008/

Reply:
This is now resolved, absolute imports are used everywhere.